apps/web/components/Badge.tsx

```diff
@@ -1,54 +1,74 @@
-import { proposalStatus, proposalTypes } from "@/types";
 import React from "react";
+import { proposalStatus, poolTypes } from "@/types";
+import {
+  CurrencyDollarIcon,
+  HandThumbUpIcon,
+} from "@heroicons/react/24/outline";
+import { capitalize } from "@/utils/text";
 
-interface BadgeProps {
-  type: number;
+type BadgeProps = {
+  poolType?: number;
+  status?: number;
+  label?: string;
   classNames?: string;
-}
+  icon?: React.ReactNode;
+  isCapitalize?: boolean;
+};
 
-interface StatusBadgeProps {
-  status: number;
-  classNames?: string;
-}
+// Styles for different pool badge types
+const POOL_TYPE_STYLES = [
+  "bg-secondary-soft text-secondary-content",
+  "bg-secondary-soft text-secondary-content",
+];
 
-// TODO!: add real styles, this is just a placeholder
-//variant for common badge
-const TYPE_STYLES = [
-  "bg-warning text-black",
-  "bg-primary text-black",
-  "bg-secondary text-white",
+// Styles for different proposal statuses badges
+const PROPOSAL_STATUS_STYLES = [
+  "bg-error-soft text-error-content",
+  "bg-primary-soft text-primary-content",
+  "bg-secondary-soft text-secondary-content",
+  "bg-error-soft text-error-content",
+  "bg-tertiary-soft text-tertiary-content",
 ];
 
-//variants for Status Badge
-const STATUS_STYLES = ["badge-error", "bg-success", "bg-warning"];
+const BASE_STYLES = "border-none rounded-full leading-5 py-2 px-4 text-base";
 
-const BASE_STYLES = "badge w-28 p-4 font-semibold";
-const BASE_STYLES_STATUS =
-  "badge text-white min-w-20 p-4 text-center tracking-widest";
+export function Badge({
+  poolType,
+  status,
+  label,
+  classNames,
+  icon,
+  isCapitalize = false,
+}: BadgeProps): JSX.Element {
+  const isStatusBadge = status !== undefined;
 
-export function Badge({ type, classNames }: BadgeProps) {
-  return (
-    <>
-      <span
-        className={`${
-          TYPE_STYLES[type] ?? "bg-accent text-black"
-        } ${BASE_STYLES} ${classNames}`}
-      >
-        {proposalTypes[type] ?? "no type"}
-      </span>
-    </>
-  );
-}
+  // Determine the appropriate styles based on whether it's a proposal status badge or a pool type badge
+  const styles = isStatusBadge
+    ? `${PROPOSAL_STATUS_STYLES[status!] ?? "bg-secondary-soft"}`
```

Comment:
status can be undefined regarding the props type, same for poolType so bang operator (!) might hide a potential problem